pyquil/api/_quantum_computer.py

```diff
@@ -129,28 +129,26 @@ def to_compiler_isa(self) -> CompilerISA:
         """
         return self.compiler.quantum_processor.to_compiler_isa()
 
-    @_record_call
     def run(
         self,
         executable: QuantumExecutable,
-        memory_map: Optional[Mapping[str, Sequence[Union[int, float]]]] = None,
     ) -> np.ndarray:
         """
-        Run a quil executable. If the executable contains declared parameters, then a memory
-        map must be provided, which defines the runtime values of these parameters.
+        Run a quil executable. All parameters in the executable must have values applied using
+        ``Program#with_parameter_values`` or ``Program#set_parameter_value``.
 
-        :param executable: The program to run. You are responsible for compiling this first.
-        :param memory_map: The mapping of declared parameters to their values. The values
-            are a list of floats or integers.
+        :param executable: The program to run, compiled as needed for its target QAM.
         :return: A numpy array of shape (trials, len(ro-register)) that contains 0s and 1s.
         """
-        self.qam.load(executable)
-        if memory_map:
-            for region_name, values_list in memory_map.items():
-                self.qam.write_memory(region_name=region_name, value=values_list)
-        return self.qam.run().wait().read_memory(region_name="ro")  # type: ignore
+        result = self.qam.run(executable)
+
+        # QUESTION (for @ameyer) - this is consistent with the V2 API, but in that API,
+        # QuantumComputer#run and QAM#run return different shapes.
```

Comment:
I think the presumption that the user wants to read out `ro` is a bit inconsistent with the method name `run`.

* If we use the name `run`, it should just return the result.
* If we want to return the measurements of a specific region, I would rename to `run_and_read_out(self, executable: QuantumExecutable, region_name: str = 'ro')`